### PR TITLE
Brianma/fi

### PR DESCRIPTION
**Pull request by martinb35**
integrate changes from windowsai into layer_dev

### CODE REVIEW

**Comment by ryanlai2** [comment]
Needs to be updated with fixes to https://github.com/microsoft/onnxruntime/pull/2469 :)

File: onnxruntime/core/graph/schema_registry.cc [hunk 1]
```diff
@@ -194,6 +194,23 @@ DomainToVersionMap SchemaRegistryManager::GetLatestOpsetVersions(bool is_onnx_on
   return domain_version_map;
 }
 
+static bool IsDomainVersionBeyondSupportedRange(
+    const std::string& domain,
+    const int op_set_version) {
+  // check the ONNX schema registry
+  auto& onnx_domain_version_map =
```
Comment (yuslepukhin) [inline onnxruntime/core/graph/schema_registry.cc]: > auto& [](start = 1, length = 6)

Prefer to specify constness explicitly even with auto

File: onnxruntime/core/graph/model.cc [hunk 2]
```diff
@@ -91,6 +91,10 @@ Model::Model(std::unique_ptr<ModelProto> model_proto, const IOnnxRuntimeOpSchema
         " specifies which version of the ONNX OperatorSet is being imported.");
   }
 
+  if (!model_proto->has_ir_version() || model_proto->ir_version() > ONNX_NAMESPACE::Version::IR_VERSION) {
+      throw std::invalid_argument("Unknown model file format version.");
```
Comment (yuslepukhin) [inline onnxruntime/core/graph/model.cc]: > throw s [](start = 5, length = 8)

ORT_THROW

File: onnxruntime/core/graph/model.cc [hunk 3]
```diff
@@ -91,6 +91,10 @@ Model::Model(std::unique_ptr<ModelProto> model_proto, const IOnnxRuntimeOpSchema
...
+  }
```
Comment (yuslepukhin) [inline onnxruntime/core/graph/model.cc]: I don't think this is inline with our requirements.

File: onnxruntime/core/graph/schema_registry.cc (hunk 1 above)
Comment (martinb35) [reply onnxruntime/core/graph/schema_registry.cc]: this is from change ([https://github.com/microsoft/onnxruntime/pull/2413](https://github.com/microsoft/onnxruntime/pull/2413)) and we are just pulling in the latest version to our branch. The change is already in master. If it needs to be addressed, please open a bug to the person who changed this.

---
In reply to: [350341673](https://github.com/microsoft/onnxruntime/pull/2470#discussion_r350341673) [](ancestors = 350341673)

File: onnxruntime/core/graph/model.cc (hunk 3 above)
Comment (martinb35) [reply onnxruntime/core/graph/model.cc]: this is from change ([https://github.com/microsoft/onnxruntime/pull/2413](https://github.com/microsoft/onnxruntime/pull/2413)) and we are just pulling in the latest version to our branch. The change is already in master. If it needs to be addressed, please open a bug to the person who changed this.

---
In reply to: [350342604](https://github.com/microsoft/onnxruntime/pull/2470#discussion_r350342604) [](ancestors = 350342604)

Comment (fdwr) [reply onnxruntime/core/graph/model.cc]: @martinb35 : It's not in master yet. I'm happy to add const, but @yuslepukhin , I'm curious about what you mean regarding ORT requirements? Have you talked with Pranav and George Wu?

**Pull request merged**